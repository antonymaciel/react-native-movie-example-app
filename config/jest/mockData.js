export const movie = {
    id: 0,
    year: '2019',
    imageUrl: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    title: 'Spiderman',
    vote_average: '9.1',
    urlbackdropImage: 'https://image.tmdb.org/t/p/w500/kqjL17yufvn9OVLyXYpvtyrFfak.jpg', 
    generes: 'Aventure, Drama, Terror',  
    vote_count: '1200',
    overview: 'Amazing movie',
    backdrop_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
    genres: ['Adventure', 'Terror', 'Drama'],
    release_date: '2016-5-14'
}

export const review = {
    id: 0,
    author: 'Antony Maciel',
    url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
    content: 'Great Movie!, always has amazing content \n I love this movie'
}

export const movies = [
    {
        id: 1,
        poster_path: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'XMan',
        vote_average: 7.8,
        release_date: '2016-5-14'
    },
    {     
        id: 2,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Superman',
        vote_average: 0.9,
        release_date: '2015-5-9'
    },
    {
        id: 3,
        year: '2019',
        imageUrl: '/kqjL17yufvn9OVLyXYpvtyrFfak.jpg',
        title: 'Spiderman',
        vote_average: 8.0,
        release_date: '2005-5-27'
    }
]


export const reviews = [
    {
        id: 1,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!!!'
    },
    {
        id: 2,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: 'Great Movie!, always has amazing content \n I love this movie'
    },
    {
        id: 3,
        author: 'Antony Maciel',
        url: 'https://www.themoviedb.org/review/5d2ddc7d6a300b0011a469df',
        content: '&^%$#@!'
    }
]

