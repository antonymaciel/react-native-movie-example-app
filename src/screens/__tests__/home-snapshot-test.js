import React from 'react';
import renderer from 'react-test-renderer';
import { movies } from '../../../config/jest/mockData';
import Home from '../home';
jest.mock('../../containers/movieComponentContainer', () => 'MovieComponentContainer');


it('renders home screen using Snapshots', () => {
  expect(renderer.create(
    <Home
        navigation={jest.fn()}
        onNewPage={jest.fn()}
        movies={movies}
    />
  )).toMatchSnapshot();
});

