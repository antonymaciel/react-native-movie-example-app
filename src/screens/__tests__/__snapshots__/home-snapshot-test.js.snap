// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders home screen using Snapshots 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    ListFooterComponent={[Function]}
    ListHeaderComponent={[Function]}
    data={
      Array [
        Object {
          "id": 1,
          "poster_path": "/kqjL17yufvn9OVLyXYpvtyrFfak.jpg",
          "release_date": "2016-5-14",
          "title": "XMan",
          "vote_average": 7.8,
        },
        Object {
          "id": 2,
          "imageUrl": "/kqjL17yufvn9OVLyXYpvtyrFfak.jpg",
          "release_date": "2015-5-9",
          "title": "Superman",
          "vote_average": 0.9,
          "year": "2019",
        },
        Object {
          "id": 3,
          "imageUrl": "/kqjL17yufvn9OVLyXYpvtyrFfak.jpg",
          "release_date": "2005-5-27",
          "title": "Spiderman",
          "vote_average": 8,
          "year": "2019",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    keyboardShouldPersistTaps="always"
    maxToRenderPerBatch={10}
    numColumns={2}
    onContentSizeChange={[Function]}
    onEndReached={[Function]}
    onEndReachedThreshold={0.2}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    refreshControl={
      <RefreshControlMock
        onRefresh={[Function]}
        refreshing={false}
      />
    }
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
      >
        <View />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <MovieComponentContainer
            movie={
              Object {
                "id": 1,
                "poster_path": "/kqjL17yufvn9OVLyXYpvtyrFfak.jpg",
                "release_date": "2016-5-14",
                "title": "XMan",
                "vote_average": 7.8,
              }
            }
            navigation={[MockFunction]}
          />
          <MovieComponentContainer
            movie={
              Object {
                "id": 2,
                "imageUrl": "/kqjL17yufvn9OVLyXYpvtyrFfak.jpg",
                "release_date": "2015-5-9",
                "title": "Superman",
                "vote_average": 0.9,
                "year": "2019",
              }
            }
            navigation={[MockFunction]}
          />
        </View>
        <View />
      </View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <MovieComponentContainer
            movie={
              Object {
                "id": 3,
                "imageUrl": "/kqjL17yufvn9OVLyXYpvtyrFfak.jpg",
                "release_date": "2005-5-27",
                "title": "Spiderman",
                "vote_average": 8,
                "year": "2019",
              }
            }
            navigation={[MockFunction]}
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
      >
        <View />
      </View>
    </View>
  </RCTScrollView>
</View>
`;
